modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/Fields.tsx

```diff
@@ -0,0 +1,493 @@
+};
+
+type FieldType = {
+	name: string;
+	selected: boolean;
+	type: string;
+	visible: boolean;
+};
+
+interface AddFDSFieldsModalContentInterface {
+	closeModal: Function;
+	fdsView: FDSViewType;
+	onSave: Function;
+}
+
+const AddFDSFieldsModalContent = ({
+	closeModal,
+	fdsView,
+	onSave,
+}: AddFDSFieldsModalContentInterface) => {
+	const [fields, setFields] = useState<Array<FieldType> | null>(null);
+	const [query, setQuery] = useState('');
+
+	const onSearch = (query: string) => {
+		setQuery(query);
+
+		if (!fields) {
+			return;
+		}
+
+		const regexp = new RegExp(query, 'i');
+
+		setFields(
+			fields.map((field) => ({
+				...field,
+				visible: Boolean(field.name.match(regexp)),
+			}))
+		);
+	};
+
+	useEffect(() => {
+		const getFields = async () => {
+			const {restApplication, restSchema} = fdsView[
+				OBJECT_RELATIONSHIP.FDS_ENTRY_FDS_VIEW
+			];
+
+			const response = await fetch(`/o${restApplication}/openapi.json`);
+
+			if (!response.ok) {
+				openToast({
+					message: Liferay.Language.get(
+						'your-request-failed-to-complete'
+					),
+					type: 'danger',
+				});
+
+				return null;
+			}
+
+			const responseJSON = await response.json();
+
+			const properties =
+				responseJSON?.components?.schemas[restSchema]?.properties;
+
+			if (!properties) {
+				openToast({
+					message: Liferay.Language.get(
+						'your-request-failed-to-complete'
+					),
+					type: 'danger',
+				});
+
+				return null;
+			}
+
+			const fieldsArray: Array<FieldType> = [];
+
+			const isObjectSchema =
+				responseJSON.components.schemas[restSchema].xml.name ===
+				'ObjectEntry';
+
+			Object.keys(properties).forEach((propertyKey) => {
+				const propertyValue = properties[propertyKey];
+
+				if (isObjectSchema && !propertyValue.extensions) {
+					return;
+				}
+
+				if (propertyKey === 'x-class-name') {
+					return;
+				}
+
+				const type = propertyValue.type;
+
+				if (type === 'object' || type === 'array') {
+					return;
+				}
+
+				fieldsArray.push({
+					name: propertyKey,
+					selected: false,
+					type,
+					visible: true,
+				});
+			});
+
+			setFields(fieldsArray);
+		};
+
+		getFields();
+	}, [fdsView]);
+
+	const isSelectAllChecked = () => {
+		if (!fields) {
+			return false;
+		}
+
+		const selectedFieldsCount =
+			fields.filter((field) => field.selected)?.length || 0;
+
+		return selectedFieldsCount === fields.length;
+	};
+
+	const isSelectAllIndeterminate = () => {
+		if (!fields) {
+			return false;
+		}
+
+		const selectedFieldsCount =
+			fields.filter((field) => field.selected)?.length || 0;
+
+		return selectedFieldsCount > 0 && selectedFieldsCount < fields.length;
+	};
+
+	const visibleFields = fields?.filter((field) => field.visible) ?? [];
+
+	return (
+		<div className="fds-view-fields-modal">
+			<ClayModal.Header>
+				{Liferay.Language.get('add-fields')}
+			</ClayModal.Header>
+
+			<ClayModal.Body>
+				{fields === null ? (
+					<ClayLoadingIndicator />
+				) : (
+					<>
```

Comment:
nit: ternary in JSX moves indentation over one step, it might be a bit easier to read if you break it up into separate conditions

```jsx
<>
    {fields === null &&
        <ClayLoadingIndicator />
    }

    {fields !== null &&
        <ManagementToolbar.Container>
         // ....
       </ManagementToolbar.Container>
    }
<>
```

Reply:
actually maybe I am wrong here.

Reply:
Yeah, indentation is not better in split code, but it is arguably more readable. For me, it's subjective and depends on the case, ternary is better the simpler the code is. For loading or empty state, I think it's fine. I only know I highly dislike ternary within a ternary.